Always enable probabilistic sandbox state flag for testing

pvollan opened the pull request:
#### cb31f36ddcbe15d3dad976c353b8a704d9683391
<pre>
Always enable probabilistic sandbox state flag for testing
<a href="https://bugs.webkit.org/show_bug.cgi?id=246762">https://bugs.webkit.org/show_bug.cgi?id=246762</a>
rdar://101349573

Reviewed by Sihui Liu.

Always enable probabilistic sandbox state flag for testing, in order to avoid flakiness of the tests
caused by the probabilistic enablement of sandbox features.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:

Canonical link: <a href="https://commits.webkit.org/255826@main">https://commits.webkit.org/255826@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/df61ddcd7c2c2f1bfd398a52ab170dcdef3cf7a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93651 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/2846 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/24283 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103302 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/2857 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31124 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86003 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/99364 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99313 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2031 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80095 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29065 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/83955 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/83701 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72017 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/event-listener (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37509 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/17540 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35351 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18800 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39227 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/41331 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1891 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41162 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38031 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (commented):
LGTM.Do we need this in TestWebKitAPI too?

szewai commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -959,7 +960,7 @@ WebPage::WebPage(PageIdentifier pageID, WebPageCreationParameters&& parameters)
 #if USE(APPLE_INTERNAL_SDK)
     uint64_t bootTime = mach_boottime_usec();
-    if (!(bootTime & 0x7)) {
+    if (!(bootTime & 0x7) || CocoaApplication::isWebkitTestRunner() || CocoaApplication::isTestWebKitAPI()) {
```
There is a `isRunningTest` in DefaultWebBrowserChecks.h. Maybe we can use that?

pvollan replied:
> There is a `isRunningTest` in DefaultWebBrowserChecks.h. Maybe we can use that?

Ah, that is a great point, will fix.

Thanks for reviewing!

pvollan committed:
Always enable probabilistic sandbox state flag for testing
https://bugs.webkit.org/show_bug.cgi?id=246762
rdar://101349573

Reviewed by Sihui Liu.

Always enable probabilistic sandbox state flag for testing, in order to avoid flakiness of the tests
caused by the probabilistic enablement of sandbox features.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:

Canonical link: https://commits.webkit.org/255826@main
